### PR TITLE
Fix golden file tests for CR mapping

### PR DESCRIPTION
**Pull request by tuommaki**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -46,7 +46,6 @@ require (
 	github.com/spf13/afero v1.3.1
 	github.com/spf13/viper v1.6.3
 	golang.org/x/sync v0.0.0-20190911185100-cd5d95a43a6e
-	gopkg.in/yaml.v2 v2.3.0
```
Comment (tuommaki) [inline go.mod:49]: Burn it with fire. :fire:

File: integration/test/crmapping/testdata/case-0-create-AzureConfig_AzureCluster.golden [hunk 2]
```diff
@@ -22,7 +22,7 @@ spec:
     vnet:
       name: c6fme-VirtualNetwork
       resourceGroup: c6fme
-  resourceGroup: ""
+  resourceGroup: c6fme
```
Comment (tuommaki) [inline integration/test/crmapping/testdata/case-0-create-AzureConfig_AzureCluster.golden:25]: :see_no_evil:

File: integration/test/crmapping/testdata/case-0-migrate-AzureConfig_AzureConfig.golden [hunk 3]
```diff
@@ -43,21 +38,24 @@ spec:
       cidr: 10.10.0.0/16
       masterSubnetCIDR: 10.10.0.0/24
       workerSubnetCIDR: 10.10.1.0/24
-    workers: null
+    workers:
+    - dockerVolumeSizeGB: 50
+      kubeletVolumeSizeGB: 100
+      vmSize: Standard_A2_v2
```
Comment (tuommaki) [inline integration/test/crmapping/testdata/case-0-migrate-AzureConfig_AzureConfig.golden:44]: Nothing to see here. It never worked. :eyes:

File: integration/test/crmapping/testdata/case-0-migrate-AzureConfig_AzureConfig.golden [hunk 4]
```diff
@@ -68,38 +66,42 @@ spec:
       cloudProvider: azure
       dns:
         ip: 172.31.0.10
-      domain: ""
+      domain: cluster.local
```
Comment (tuommaki) [inline integration/test/crmapping/testdata/case-0-migrate-AzureConfig_AzureConfig.golden:69]: Ok. I admit. I have no idea what I'm doing.

**Review by whites11 — APPROVED** [review]
Looks ok when tested

**Review by tuommaki — COMMENTED** [review]
I wasn't here

File: service/controller/resource/capzcrs/create_test.go [hunk 5]
```diff
@@ -106,6 +106,9 @@ func Test_AzureConfigCRMapping(t *testing.T) {
 			r := constructCapzcrsHandler(t, client, tc.location)
 
 			err = r.EnsureCreated(context.Background(), azureConfig)
+			if err != nil {
```
Comment (fiunchinho) [inline service/controller/resource/capzcrs/create_test.go:109]: mm... I believe this should not be here, so you can use the switch/case later and assert expected errors.

**Pull request merged**